### PR TITLE
docs(recipe): Provide recipe for integration with RxAngular

### PR DESCRIPTION
**Pull request by Carniatto**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngxs/store/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[x] Other... Please describe: Documentation
```

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by joaqcid — APPROVED** [review]

File: docs/recipes/intregration-with-rxangular.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+};
+
+@Component({
+  selector: 'app-heroes',
+  templateUrl: './heroes.component.html',
+  styleUrls: ['./heroes.component.css'],
+  providers: [RxState] // <--- This binds RxState to the lifetime of the component
+})
+export class HeroesComponent {
+  @Select(getHeroes) SateHeroes$: Obersavable<Hero[]>; // <--- normal @Select use from NGXS
```
Comment (sebaferreras) [inline docs/recipes/intregration-with-rxangular.md:24]: @Carniatto Super minor but I think there're two typos in this line :)

SateHeroes => S**t**ateHeroes
Obersavable => Ob**ser**vable

Comment (arturovt) [reply docs/recipes/intregration-with-rxangular.md:24]: StateHeroes => stateHeroes :smile:

**Comment by Carniatto** [comment]
@markwhitfeld can we merge this?

File: docs/recipes/intregration-with-rxangular.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
+# Integration with RxAngular
+There are use cases when is desired to have a transient component state has it's lifetime bound to the component lifecycle.
+[RxAngular](https://github.com/rx-angular/rx-angular) provides a solution for this problem that is fully reactive and with focus on runtime performance and template rendering.
+
+In order to leverage this library and all the power of a global state management is possible to integrate RxAngular with NGXS.
+First let's bind a value provinient from NGXS state to RxAngular component state:
```
Comment (markwhitfeld) [inline docs/recipes/intregration-with-rxangular.md:6]: "provinient"?
Not sure what this means?

File: docs/recipes/intregration-with-rxangular.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  readonly add = new Subject<string>();
+  readonly delete = new Subject<Hero>();
+
+  constructor(
+    private state: RxState<HeroesComponentState>,
+    private store: Store
+  ) {
+
+    this.state.hold( // <--- RxAngular hold will manage the subscription for us
+      this.add.pipe(switchMap(name => this.store.dispatch(new AddHero(name)))) <--- dispatch action to NGXS
```
Comment (markwhitfeld) [inline docs/recipes/intregration-with-rxangular.md:61]:
```suggestion
      this.add.pipe(switchMap(name => this.store.dispatch(new AddHero(name)))) // <--- dispatch action to NGXS
```

File: docs/recipes/intregration-with-rxangular.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    );
+
+    this.state.hold(
+      this.delete.pipe(
+        switchMap(hero => this.stor(new DeleteHero(hero))e.dispatch)
```
Comment (markwhitfeld) [inline docs/recipes/intregration-with-rxangular.md:66]:
```suggestion
        switchMap(hero => this.store.dispatch(new DeleteHero(hero)))
```

File: docs/recipes/intregration-with-rxangular.md (hunk 1 above)
Comment (markwhitfeld) [inline docs/recipes/intregration-with-rxangular.md:24]:
```suggestion
  @Select(getHeroes) stateHeroes$: Observable<Hero[]>; // <--- normal @Select use from NGXS
```

**Review by markwhitfeld — CHANGES_REQUESTED** [review]
A couple of fixes needed.
Please also rebase your changes onto `release` this will prevent the additional commits in your PR.

I also think that the example should include how to combine the state from NGXS with the state held within RxAngular.

File: docs/recipes/intregration-with-rxangular.md (hunk 2 above)
Comment (Carniatto) [reply docs/recipes/intregration-with-rxangular.md:6]: it was a typo actually 
```suggestion
First, let's bind a value from NGXS state to RxAngular component state: 
```

File: docs/recipes/intregration-with-rxangular.md (hunk 1 above)
Comment (Carniatto) [reply docs/recipes/intregration-with-rxangular.md:24]:
```suggestion
  @Select(getHeroes) stateHeroes$: Observable<Hero[]>; // <--- normal @Select use from NGXS
```

**Review by markwhitfeld — APPROVED** [review]

**Comment by markwhitfeld** [comment]
This still needs to be added to the example, but I'll merge this PR for now.
You can do a separate PR to address: 
> I also think that the example should include how to combine the state from NGXS with the state held within RxAngular.

**Pull request merged**